src/java/org/apache/cassandra/index/sai/IndexContext.java

```diff
@@ -690,8 +687,10 @@ public void invalidate(boolean obsolete)
         dropped = true;
         liveMemtables.clear();
         viewManager.invalidate(obsolete);
-        indexMetrics.release();
-        columnQueryMetrics.release();
+        if (indexMetrics != null)
+            indexMetrics.release();
+        if (columnQueryMetrics != null)
```

Comment:
How can that be ever null? It's initialized in the constructor.
Maybe we can add `@Nonnull` to the field and avoid the check?